release-26.2: sql: TestIndexBackfillAfterGC failed

sql.TestIndexBackfillAfterGC [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21320664?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21320664?buildTab=artifacts#/) on release-26.2 @ [7dad80aa9d43bf14684b9b3dd68397207a4f2f30](https://github.com/cockroachdb/cockroach/commits/7dad80aa9d43bf14684b9b3dd68397207a4f2f30):


```
=== RUN   TestIndexBackfillAfterGC
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/230b14ff9b70e2729ccc9b9deabcc166/logTestIndexBackfillAfterGC3608008020
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:638: DRPC is enabled (via testserver factory defaults)
    schema_changer_test.go:4237: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/230b14ff9b70e2729ccc9b9deabcc166/logTestIndexBackfillAfterGC3608008020
--- FAIL: TestIndexBackfillAfterGC (78.60s)
=== RUN   TestIndexBackfillAfterGC/useDeclarative=false
    schema_changer_test.go:4219: pq: polling for queued jobs to complete: poll-show-jobs: batch timestamp 1776737632.411025461,0 must be after replica GC threshold 1776737632.411758827,0 (r83: /Tenant/10{-/Table/106/2})
--- FAIL: TestIndexBackfillAfterGC/useDeclarative=false (0.82s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestIndexBackfillAfterGC.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63102

--- a/pkg/sql/schema_changer_test.go
+++ b/pkg/sql/schema_changer_test.go
@@ -4212,6 +4212,17 @@ func TestIndexBackfillAfterGC(t *testing.T) {
 		sqlDB.Exec(t, `CREATE TABLE t.test (k INT PRIMARY KEY, v INT, pi DECIMAL DEFAULT (DECIMAL '3.14'))`)
 		sqlDB.Exec(t, `INSERT INTO t.test VALUES (1, 1)`)
 
+		// Split t.test off into its own range. Without this, t.test shares a
+		// range with system tables (notably system.jobs), and the GCRequest
+		// injected below would advance the GC threshold on that shared range.
+		// The schema changer recovers from BatchTimestampBeforeGCError on the
+		// backfill, but the parallel poll-show-jobs query in jobs.WaitForJobs
+		// would race with the threshold bump and fail the CREATE INDEX.
+		tableDesc := desctestutils.TestingGetPublicTableDescriptor(kvDB, codec, "t", "test")
+		if err := kvDB.AdminSplit(ctx, codec.TablePrefix(uint32(tableDesc.GetID())), hlc.MaxTimestamp); err != nil {
+			t.Fatal(err)
+		}
+
 		shouldRunGC.Store(true)
 		if _, err := db.Exec(
 			fmt.Sprintf(`CREATE UNIQUE INDEX %s ON t.test (v)`, indexName),

